### PR TITLE
Split unit and e2e tests

### PR DESCRIPTION
**Pull request by msm-code**
Unit tests should be fast to execute, while e2e tests are allowed to
contact the real service. E2e tests are too slow to run locally,
but we can still run them in the CI.

**Your checklist for this pull request**
- [x] I've ~~read~~ written the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [x] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)
- [x] I've added automated tests for my change (if applicable, optional)
- [x] I've updated documentation to reflect my change (if applicable)

**What is the current behaviour?**
unit tests and e2e tests in one pipeline stage and directory

**What is the new behaviour?**
They are in separate folders

**Test plan**
Wait for the pipeline for this change to succeed

_After submitting, your code will be tested by the CI pipeline. Please
ensure that all tests pass. If they don't at first, please update your code_

IKR

**Closing issues**

N/A

### CODE REVIEW

File: src/tests/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
 pytest
 pytest-timeout
-requests
 zmq
```
Comment (icedevml) [inline src/tests/requirements.txt:3]: dependency pinning?

File: src/e2etests/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.6
+RUN apt update; apt install -y cmake
+RUN pip3 install pytest
```
Comment (icedevml) [inline src/e2etests/Dockerfile:3]: maybe `pytest==5.4.1`?

File: .github/workflows/test_code.yml [hunk 3]
```diff
@@ -83,16 +83,29 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
       - name: build test image
         run: docker build -t mquery_tests:latest -f src/tests/Dockerfile .
+      - name: run unit tests
+        run: docker run mquery_tests
+      - name: stop docker compose
+        if: always()
```
Comment (icedevml) [inline .github/workflows/test_code.yml:98]: what is the purpose of this?

Comment (msm-code) [reply .github/workflows/test_code.yml]: :thinking:

Comment (msm-code) [reply .github/workflows/test_code.yml]: yes

File: src/e2etests/Dockerfile (hunk 2 above)
Comment (msm-code) [reply src/e2etests/Dockerfile]: :+1:

File: src/tests/requirements.txt (hunk 1 above)
Comment (msm-code) [reply src/tests/requirements.txt]: yes

File: src/e2etests/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+pytest
+pytest-timeout
+requests
```
Comment (icedevml) [inline src/e2etests/requirements.txt:3]: hmmm, code suggestion is the most top secret GitHub feature I've ever seen

```suggestion
pytest==5.4.1
pytest-timeout==1.3.4
requests==2.23.0
```

File: src/e2etests/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# E2E tests
+
+Slow test suite, used as a sanity test for mquery matching capabilities.
+Hopefully it won't allow us to merge a completely broken version.
+
+They are automatically built and run on every commit in the CI pipeline,
```
Comment (nazywam) [inline src/e2etests/README.md:6]:
```suggestion
They are automatically built and ran on every commit in the CI pipeline,
```

**Pull request merged**